client/src/components/ProtectedRoute.js

```diff
@@ -1,19 +1,42 @@
-import React, { useContext } from "react";
+import React, { useContext, useEffect } from "react";
 import { Redirect, Route } from "react-router-dom";
-import { authContext } from "...";
+import { AuthContext } from "../context/authContext";
+import Cookie from "js-cookie";
 
-const ProtectedRoute = ({ children, ...rest }) => {
-  // Check if the user is authenticated with the useContext hook.
-  const { isAuthenticated, checkValidToken } = useContext(authContext);
+const ProtectedRoute = ({ component: Comp, path, ...rest }) => {
+  const { isAuthenticated, login, logout } = useContext(AuthContext);
 
-  // checkValidToken()
+  useEffect(() => {
+    const checkTokenValidity = async () => {
+      try {
+        const cookie = Cookie.get("x-auth-token");
+        if (!cookie) throw new Error("No valid cookie");
+        const urlencoded = new URLSearchParams();
+        urlencoded.append("token", cookie);
+        const response = await fetch("http://localhost:3001/auth/checkAuth", {
```

Comment:
this can simply be `auth/checkAuth`, no need for the url

Reply:
Interesting. I tried it and it works like you said. But I don't understand why. I thought I was in the front-end (port 3000), making a call to the back-end (port 3001), so I needed to specify the URL and port...?